### PR TITLE
add mysql replication lag detector

### PR DESCRIPTION
**Pull request by frederic-peraud**
@BzSpi @Shr3ps @xp-1000 

Add Azure Mysql Replication Lag detector based on SignalFX `seconds_behind_master` metric.

### CODE REVIEW

File: cloud/azure/mysql/detectors-mysql.tf [hunk 1]
```diff
@@ -134,3 +134,32 @@ resource "signalfx_detector" "memory_usage" {
...
+
+  program_text = <<-EOF
+        base_filter = filter('resource_type', 'Microsoft.DBforMySQL/servers') and filter('primary_aggregation_type', 'true')
+        signal = data('seconds_behind_master', filter=base_filter and ${module.filter-tags.filter_custom})${var.replication_lag_aggregation_function}.publish('signal')
+        detect(when(signal > threshold(${var.replication_lag_threshold_critical}), lasting="${var.replication_lag_timer}")).publish('CRIT')
+        detect(when(signal > threshold(${var.replication_lag_threshold_warning}), lasting="${var.replication_lag_timer}") and when(signal <= ${var.replication_lag_threshold_critical})).publish('WARN')
+    EOF
+
+  rule {
+    description           = "is too high > ${var.replication_lag_threshold_critical}"
```
Comment (xp-1000) [inline cloud/azure/mysql/detectors-mysql.tf:149]: you can add unit if you want

File: cloud/azure/mysql/detectors-mysql.tf [hunk 2]
```diff
@@ -134,3 +134,32 @@ resource "signalfx_detector" "memory_usage" {
     parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
   }
 }
+
+resource "signalfx_detector" "replication_lag" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Azure MySQL Replication Lag"
```
Comment (xp-1000) [inline cloud/azure/mysql/detectors-mysql.tf:139]: keep "replication lag" in lower case please

**Review by xp-1000 — APPROVED** [review]

**Comment by xp-1000** [comment]
keep in mind this will lead to duplicate alerting if you also deploy database/mysql module (and monitor which give more frequent metrics).

File: cloud/azure/mysql/detectors-mysql.tf (hunk 1 above)
Comment (BzSpi) [reply cloud/azure/mysql/detectors-mysql.tf]: You **must** add unit :)

File: cloud/azure/mysql/variables.tf [hunk 3]
```diff
@@ -303,3 +303,65 @@ variable "memory_usage_threshold_warning" {
...
+}
+
+variable "replication_lag_timer" {
+  description = "Transformation window for replication_lag detector (i.e. 5m, 20m, 1h, 1d)"
+  type        = string
+  default     = "15m"
+}
+
+variable "replication_lag_threshold_critical" {
+  description = "Critical threshold for replication_lag detector"
```
Comment (BzSpi) [inline cloud/azure/mysql/variables.tf:358]:
```suggestion
  description = "Critical threshold in seconds for replication_lag detector"
```

File: cloud/azure/mysql/variables.tf [hunk 4]
```diff
@@ -303,3 +303,65 @@ variable "memory_usage_threshold_warning" {
...
+  type        = number
+  default     = 200
+}
+
+variable "replication_lag_threshold_warning" {
+  description = "Warning threshold for replication_lag detector"
```
Comment (BzSpi) [inline cloud/azure/mysql/variables.tf:364]:
```suggestion
  description = "Warning threshold in seconds for replication_lag detector"
```

**Review by BzSpi — CHANGES_REQUESTED** [review]

**Review by BzSpi — APPROVED** [review]

**Pull request merged**